CMakeLists.txt

```diff
@@ -193,6 +193,7 @@ if(${IREE_ENABLE_CCACHE})
   endif()
 endif()
 
+option(IREE_DEV_MODE "Configure settings to optimize for IREE development (as opposed to CI or release)" OFF)
```

Comment:
You may want to document this in https://github.com/google/iree/blob/main/docs/developers/get_started/cmake_options_and_variables.md?

Reply:
I mean I don't *want* to, but I definitely should :stuck_out_tongue:

Reply:
As my reviews are optional anyway/don't approve anything, I tried to phrase it politely :stuck_out_tongue_winking_eye:

Reply:
Eh only because GitHub doesn't have a setting for "can merge with review/approve others' PRs for merging, but not have full write access"

Reply:
As a meta note. I think we should probably not be duplicating documentation between the CMake config and the markdown. Seems like we should be able to generate this (a task for a later time)

Reply:
> Eh only because GitHub doesn't have a setting for "can merge with review/approve others' PRs for merging, but not have full write access"

No problem at all. It's fine the way it is :)

> As a meta note. I think we should probably not be duplicating documentation between the CMake config and the markdown. Seems like we should be able to generate this (a task for a later time)

I think the docs are more expressive than the strings in the CMake config. But if we can, we should do.